### ISSUE TITLE
aws_launch_configuration.this (deposed #0): ResourceInUse: Cannot delete launch configuration because it is attached to AutoScalingGroup.

### ISSUE THREAD

**Issue by d-helios** [terraform-aws-modules/terraform-aws-autoscaling#8]
Hi,
When I try to add __associate_public_ip_address__ to launch configuration, __terraform apply__ failed with following error:
```
Error: Error applying plan:

2 error(s) occurred:

* module.zabbix_app_asg.module.launch_configuration.aws_launch_configuration.this (destroy): 1 error(s) occurred:

* aws_launch_configuration.this (deposed #0): 1 error(s) occurred:

* aws_launch_configuration.this (deposed #0): ResourceInUse: Cannot delete launch configuration zabbix-app-lc-20171125225647460700000002 because it is attached to AutoScalingGroup
	status code: 400, request id: 19197d51-d235-11e7-8d7b-19219f15fe4b
* module.bastion_ec2.aws_instance.this: aws_instance.this: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.
```

## Terraform version
```
Terraform v0.11.0
+ provider.aws v1.3.1
```

## Error Log
[autoscaling_terraform.bug.log](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/files/1503705/autoscaling_terraform.bug.log)

## Stemps to reproduce
* terraform init
* terraform apply
* uncomment __associate_public_ip_address__ properties in launch configuration
* terraform apply (Error) 

## Example
[TF-Example2.zip](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/files/1503752/TF-Example2.zip)

**Comment by antonbabenko** [comment]
Thanks for opening issues in terraform-aws-modules. There are two different issues in this one:
1. `aws_launch_configuration.this (deposed #0): ResourceInUse: Cannot delete launch configuration zabbix-app-lc-20171125225647460700000002 because it is attached to AutoScalingGroup` - this is related to this module and I will work on it during few hours today.
1. `module.bastion_ec2.aws_instance.this: aws_instance.this: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.` - let us track it [separately](https://github.com/terraform-aws-modules/terraform-aws-ec2-instance/issues/10).

**Comment by nodefortytwo** [comment]
I am having the same issue regarding the ResourceInUse problem when trying to change the AMI_ID

**Comment by antonbabenko** [comment]
https://github.com/terraform-aws-modules/terraform-aws-autoscaling/releases/tag/v2.0.0 - is released. Please read README and report about any issues you can find.

**Comment by antonbabenko** [comment]
@nodefortytwo - I believe this problem has been also fixed in the release of [v2.0.0](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/releases/tag/v2.0.0)

**Comment by d-helios** [comment]
works well for me, thanks!
